### PR TITLE
#174797463 add signup page

### PR DESCRIPTION
**Pull request by Habinezajanvier**
#### What does this PR do?
Adding the signup page to the web
#### Description of Task to be completed?
- Creating a Signup form
- Validating users input before submitting
- Routing signup from the landing page
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-frontend.git`
- Switch your working directory `cd script-heroes-frontend`
- checkout the branch `git checkout ft-user-signup-#174797463`
- Create a `.env` file and update it with data from `.env-sample`
- Install all dependencies `yarn install`
- Start the server `yarn run dev`
- Click on the signup link found in the navigation bar
- Fill in your data in the Signup form found on the page
> You should see an error if you inserted the wrong data
> You should see a successful message if your data accepted
#### What are the relevant pivotal tracker stories?
[#174797463](https://www.pivotaltracker.com/story/show/174797463)
#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Good job but check on the landing page

File: src/components/SignupForm/index.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+          </FormGroup>
+        </Col>
+      </Row>
+      <div className={styles.Response}>
+        {loading && <Spinner color="secondary" />}
+        <Alert message={message} isOpen={visible} color={color} />
+      </div>
+      <Row className={styles.Button}>
+        <Col md={6}>
+          <Button style={{ width: "100%" }}>Sign in</Button>
```
Comment (jpirumvaa) [inline src/components/SignupForm/index.js]: The button should be signup instead of sign in

**Review by jpirumvaa — APPROVED** [review]
Apart from Coverage and the sign in button instead of signup, everything else is good to go on my end.

File: src/components/SignupForm/index.js (hunk 1 above)
Comment (Habinezajanvier) [reply src/components/SignupForm/index.js]: Done, Thank you brother

**Review by valens10 — APPROVED** [review]

File: debug.log [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[1116/112430.089:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
+[1117/140550.912:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
+[1119/060149.572:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (b0nbon1) [inline debug.log]: could you add this `.gitignore`?

File: src/App.scss [hunk 3]
```diff
@@ -162,126 +165,148 @@ $mainBgColor: #FBFCFC;
...
+  margin: 0 auto;
+  margin-top: 3em;
+  width: 90%;
+  img {
+    height: 100%;
+    width: 100%;
+  }
+  h1 {
+    color: #f9a826;
+    font-size: 26pt;
```
Comment (b0nbon1) [inline src/App.scss]: on font-size I would suggest try using `rem`. research on the difference between rem, em and px

File: src/components/HomePage.js [hunk 4]
```diff
@@ -1,102 +1,112 @@
...
+import { Link } from "react-router-dom";
+import { data } from "../helpers/services";
+import destinations from "../assets/destinations.png";
+import previous from "../assets/previous.png";
+import next from "../assets/next.png";
+import Card from "./Card";
+import Navigation from "./Navigation";
+import Footer from "./Footer";
 
 class NewHomePage extends Component {
```
Comment (b0nbon1) [inline src/components/HomePage.js:12]: be a bit consistent and choose between class components and functional components.

File: src/tests/components/Alert.test.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { describe } from "@jest/globals";
+import React from "react";
+import { shallow } from "enzyme";
+import Alert from "../../components/commons/Alert";
+
+describe("<Alert/>", () => {
+  it("Should render alert correctly", () => {
+    const wrapper = shallow(<Alert message="" isOpen={true} />);
+    expect(wrapper).toMatchSnapshot();
```
Comment (b0nbon1) [inline src/tests/components/Alert.test.js]: please avoiding using toMatchSnapshot, [read this for reference ](https://medium.com/@tomgold_48918/why-i-stopped-using-snapshot-testing-with-jest-3279fe41ffb2)

File: src/tests/components/Signup.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  let store;
+
+  it("Should render signup page", () => {
+    store = mockStore(initialState);
+    let wrapper = shallow(
+      <Provider store={store}>
+        <Signup />
+      </Provider>
+    );
+    expect(wrapper).toMatchSnapshot();
```
Comment (b0nbon1) [inline src/tests/components/Signup.test.js]: please avoiding using toMatchSnapshot, [read this for reference ](https://medium.com/@tomgold_48918/why-i-stopped-using-snapshot-testing-with-jest-3279fe41ffb2)

File: src/tests/components/SignupImage.test.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { describe } from "@jest/globals";
+import React from "react";
+import { shallow } from "enzyme";
+import SignupImage from "../../components/SignupImage";
+
+describe("<Alert/>", () => {
+  it("Should render alert correctly", () => {
+    const wrapper = shallow(<SignupImage />);
+    expect(wrapper).toMatchSnapshot();
```
Comment (b0nbon1) [inline src/tests/components/SignupImage.test.js]: please avoiding using toMatchSnapshot, [read this for reference ](https://medium.com/@tomgold_48918/why-i-stopped-using-snapshot-testing-with-jest-3279fe41ffb2)

File: webpack.config.js [hunk 8]
```diff
@@ -1,65 +1,71 @@
...
   },
   plugins: [
+    new Dotenv({
+      path: ".env",
+      safe: true,
+      allowEmptyValues: false,
+    }),
     new HtmlWebPackPlugin({
-      template: './src/index.html',
+      template: "./src/index.html",
```
Comment (b0nbon1) [inline webpack.config.js:68]: there are no eslint rules, for quotation marks?

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Pull request merged**